packages/apollo-graphql/src/schema/buildSchemaFromSDL.ts

```diff
@@ -216,6 +221,38 @@ export function addResolversToSchema(
       }
     }
 
+    if (isScalarType(type)) {
+      for (const fn in fieldConfigs) {
+        (type as any)[fn] = (fieldConfigs as any)[fn];
+      }
+    }
+
+    if (isEnumType(type)) {
+      const values = type.getValues();
+      const newValues = {};
+      values.forEach(value => {
+        const newValue = (fieldConfigs as any)[value.name] || value.name;
+        (newValues as any)[value.name] = {
+          value: newValue,
+          deprecationReason: value.deprecationReason,
+          description: value.description,
+          astNode: value.astNode
+        };
+      });
+
+      // In place updating hack to get around pulling in the full
+      // schema walking and immutable updating machinery from graphql-tools
+      Object.assign(
```

Comment:
@trevor-scheer what do you think of this? At first I tried copying direct from grapql-tools, but it ended up requiring a lot more dependencies than I expected. (https://github.com/apollographql/graphql-tools/blob/master/src/transforms/ConvertEnumValues.ts#L10, which requires https://github.com/apollographql/graphql-tools/blob/master/src/transforms/visitSchema.ts#L38, which requires https://github.com/apollographql/graphql-tools/blob/master/src/stitching/schemaRecreation.ts#L27, etc.)